controllers/edgedevicesignedrequest_controller.go

```diff
@@ -0,0 +1,143 @@
+//+kubebuilder:rbac:groups=management.project-flotta.io,resources=edgedevices/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=management.project-flotta.io,resources=edgedevicesignedrequest,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=management.project-flotta.io,resources=edgedevicesignedrequest/status,verbs=get;update;patch
+
+//Reconcile each edgedevicesignedrequest
+func (r *EdgeDeviceSignedRequestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling", "edgedeviceSignedRequest", req)
+
+	edsr, err := r.EdgedeviceSignedRequestRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	if !edsr.Spec.Approved {
+		if IsPending(edsr) && len(edsr.Status.Conditions) > 0 {
+			return ctrl.Result{}, nil
+		}
+		if err := r.markStatusPending(ctx, edsr); err != nil {
+			logger.Error(err, "cannot patch status to pending")
+			return ctrl.Result{Requeue: true}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	_, err = r.EdgeDeviceRepository.Read(ctx, edsr.Name, edsr.Spec.TargetNamespace)
+	if err == nil {
+		// device is already created
+		if err := r.markStatusApproved(ctx, edsr); err != nil {
+			logger.Error(err, "cannot patch status to approved")
+			return ctrl.Result{Requeue: true}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	now := v1.Now()
+	device := &v1alpha1.EdgeDevice{
+		TypeMeta: v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      edsr.Name,
+			Namespace: edsr.Spec.TargetNamespace,
+			Labels: map[string]string{
+				v1alpha1.EdgeDeviceSignedRequestLabelName: v1alpha1.EdgeDeviceSignedRequestLabelValue,
+			},
+		},
+		Spec: managementv1alpha1.EdgeDeviceSpec{
+			RequestTime: &now,
+		},
+	}
+	err = r.EdgeDeviceRepository.Create(ctx, device)
+	if err != nil {
+		logger.Error(err, "cannot create edgedevice")
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	if err := r.markStatusApproved(ctx, edsr); err != nil {
+		logger.Error(err, "cannot patch status to approved")
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *EdgeDeviceSignedRequestReconciler) markStatusPending(ctx context.Context, edsr *v1alpha1.EdgeDeviceSignedRequest) error {
+	patch := client.MergeFrom(edsr.DeepCopy())
+	now := v1.NewTime(time.Now())
+	message := "device is waiting for approval"
+	edsr.Status.Conditions = []v1alpha1.EdgeDeviceSignedRequestCondition{
+		{
+			Type:               v1alpha1.EdgeDeviceSignedRequestStatusPending,
+			Status:             "true",
+			Message:            &message,
+			LastTransitionTime: &now,
+		},
+	}
+	err := r.EdgedeviceSignedRequestRepository.PatchStatus(ctx, edsr, &patch)
+	return err
+}
+
+func (r *EdgeDeviceSignedRequestReconciler) markStatusApproved(ctx context.Context, edsr *v1alpha1.EdgeDeviceSignedRequest) error {
```

Comment:
List of historical conditions should be maintained and if a status/message/reason of an existing condition needs to be changed, it should be changed in the object present in the `Conditions` slice or added to it otherwise.